### ISSUE TITLE
flb_utils.c:339 errno=34] Result too large

### ISSUE THREAD

**Issue by smil2k** [fluent/fluent-bit#2776]
## Bug Report

**Describe the bug**
Fluentbit for windows stops sometimes with an error in the title.
It seems that "recv" in the flb_utils_timer_consume() would like to return more data 

```
[2020/11/13 16:49:48] [ info] [output:azure:azure.1] customer_id=..., HTTP status=200
[2020/11/13 16:49:58] [ info] [output:azure_blob:azure_blob.0] content appended to blob successfully
[2020/11/13 16:49:58] [ info] [output:azure:azure.1] customer_id=..., HTTP status=200
[2020/11/13 16:51:27] [error] [C:\projects\fluent-bit-2e87g\src\flb_utils.c:339 errno=34] Result too large
[2020/11/13 16:51:27] [ warn] [engine] service will stop in 5 seconds
[2020/11/13 16:51:32] [ info] [engine] service stopped
[2020/11/13 16:51:33] [ info] [engine] started (pid=10028)
�

[2020/11/13 16:51:38] [ info] [output:azure:azure.1] customer_id=..., HTTP status=200
[2020/11/13 16:51:38] [ info] [output:azure:azure.1] customer_id=..., HTTP status=200
[2020/11/13 16:51:38] [ info] [output:azure_blob:azure_blob.0] content appended to blob successfully
[2020/11/13 16:51:55] [error] [C:\projects\fluent-bit-2e87g\src\flb_utils.c:339 errno=34] Result too large
[2020/11/13 16:51:55] [error] [C:\projects\fluent-bit-2e87g\src\flb_utils.c:339 errno=34] Result too large
[2020/11/13 16:51:55] [ warn] [engine] service will stop in 5 seconds
[2020/11/13 16:52:00] [ info] [engine] service stopped
[2020/11/13 16:52:01] [ info] [engine] started (pid=9340)
�

[2020/11/13 16:53:51] [ info] [output:azure_blob:azure_blob.0] content appended to blob successfully
[2020/11/13 16:53:51] [ info] [output:azure:azure.1] customer_id=.., HTTP status=200
[2020/11/13 16:53:56] [ info] [output:azure_blob:azure_blob.0] content appended to blob successfully
[2020/11/13 16:53:56] [ info] [output:azure:azure.1] customer_id=..., HTTP status=200
[2020/11/13 16:53:58] [error] [C:\projects\fluent-bit-2e87g\src\flb_utils.c:339 errno=34] Result too large
[2020/11/13 16:53:58] [ warn] [engine] service will stop in 5 seconds
[2020/11/13 16:54:03] [ info] [engine] service stopped
[2020/11/13 16:54:04] [ info] [engine] started (pid=10640)
[2020/11/13 16:54:04] [ info] [storage] version=1.0.6, initializing...
�
```

**To Reproduce**

I was not able to reproduce it in our dev environment. Sometimes we have extra long log lines, they are skipped by config.

**Your Environment**

* Version used: 1.6.1
* Configuration: 

```ini
[SERVICE]
    Parsers_File C:\fluent-bit-1.6.1\conf\parsers.conf
    Log_File C:\fluent-bit-1.6.1\conf\fluent.log

[INPUT]
    Name tail
    Path C:\Users\*\.powerclient.v7\logs\all.log
    Path_Key logFilePath
    Multiline on
    Parser_Firstline clientlog
    DB C:\fluent-bit-1.6.1\fluent-bit.db
    Ignore_Older 1d
    Skip_Long_Lines true
    Tag *

@INCLUDE C:\fluent-bit-1.6.1\conf\filters.conf

[OUTPUT]
    Name azure_blob
    Match *
    account_name ...
    shared_key ...
    container_name clientlogs
    auto_create_container off
    tls on

[OUTPUT]
    Name azure
    Match *
    Customer_ID ....
    Shared_Key ...

* Operating System and version: Windows Server 2016 Datacenter edition
* Filters and plugins:

[FILTER]
    Name nest
    Match tail*
    Operation lift
    Nested_under attributes

[FILTER]
    Name record_modifier
    Match tail*
    # These fields are incorrectly decoded as 2147483647, because fluentbit doesnt like 64 bit integers inside a json.
    # The fields are still present in the json
    Remove_Key start_time
    Remove_Key end_time
    Record hostname ${COMPUTERNAME}
    Record config_version 2020-11-13

[FILTER]
    Name parser
    Parser username
    Match tail*
    Key_name logFilePath
    Reserve_Data True
    Preserve_Key True

[FILTER]
    Name parser
    Parser hourly
    Match tail*
    Key_name log_datetime
    Reserve_Data True
    Preserve_Key True

[FILTER]
    Name parser
    Parser ltrim
    Match tail*
    Key_name message
    Reserve_Data True
    Preserve_Key False

[FILTER]
    Name          rewrite_tag
    Match         tail*
    Rule          message .* clientlogs.$os_user.$tmp_date.$tmp_hour.log true
    Emitter_Name  re_emitted

[FILTER]
    Name record_modifier
    Match *
    Remove_Key tmp_date
    Remove_Key tmp_hour
```

**Comment by fujimotos** [comment]
@smil2k  I'm reading your failure log. The interesting part is that Fluent Bit
seems to shutdown _gracefully_.

```
[2020/11/13 16:53:58] [ warn] [engine] service will stop in 5 seconds
[2020/11/13 16:54:03] [ info] [engine] service stopped
[2020/11/13 16:54:04] [ info] [engine] started (pid=10640)
[2020/11/13 16:54:04] [ info] [storage] version=1.0.6, initializing...
```

This suggests that someone called `fb_engne_stop()`. There are few possibilities,
but one possibility is that Windows Service Controller requested it.

Can you check the Windows Event Log on your prod instance? (e.g. executing
`get-eventlog system` on powershell). There might provide some hints on why
Fluent Bit stopped.

**Comment by smil2k** [comment]
Ok, that explains a lot.

**Comment by smil2k** [comment]
I can confirm that this is the sign of stop. I would advice to add an info level log into the log that the service was killed normally, because the normal shutdown causes errors in network services and it makes this confusing.

**Comment by fujimotos** [comment]
FWIW: I fixed the network error part in #2812.
